pkg/inspector/image-inspector.go

```diff
@@ -541,3 +485,154 @@ func createOutputDir(dirName string, tempName string) (string, error) {
+		DockerAuthConfig: nil,
+		DockerCertPath:   certPath,
+	}
+
+	if i.opts.DstPath, err = createOutputDir(i.opts.DstPath, "image-inspector-"); err != nil {
+		return nil, "", err
+	}
+	destRef, err := directory.NewReference(i.opts.DstPath)
+	if err != nil {
+		return nil, "", fmt.Errorf("Invalid destination name %s: %v", i.opts.DstPath, err)
+	}
+
+	imagePullAuths, err := i.getAuthConfigs()
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	log.Println("Copying image")
+	for name, auth := range imagePullAuths.Configs {
+		sourceCtx.DockerAuthConfig = &types.DockerAuthConfig{
+			Username: auth.Username,
+			Password: auth.Password,
+		}
+		reportBuffer := &bytes.Buffer{}
+		authErr = copy.Image(policyContext, destRef, srcRef, &copy.Options{
+			RemoveSignatures: false,
+			SignBy:           "",
+			ReportWriter:     reportBuffer,
+			SourceCtx:        sourceCtx,
+			DestinationCtx:   nil,
+			ProgressInterval: PULL_LOG_INTERVAL_SEC,
+		})
+		log.Printf("Copying image output:\n%s", reportBuffer.String())
+		if authErr == nil {
+			break
+		}
+		log.Printf("Authentication with %s failed: %v", name, authErr)
+	}
+
+	if authErr != nil {
+		return nil, "", fmt.Errorf("Unable to pull docker image: %v\n", authErr)
+	}
+
+	img, err := srcRef.NewImage(sourceCtx)
+	if err != nil {
+		return nil, "", err
+	}
+
+	log.Println("Inspecting image")
+	rawManifest, _, err := img.Manifest()
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while reading image manifest: %v\n", err)
+	}
+	imageDigest, err := manifest.Digest(rawManifest)
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while parsing image manifest: %v\n", err)
+	}
+
+	inspectInfo, err := img.Inspect()
+	if err != nil {
+		return inspectInfo, "", fmt.Errorf("Error while inspecting copied image Manifest: %v\n", err)
+	}
+
+	err = i.extractDownloadedImage(inspectInfo.Layers)
+	if err != nil {
+		return nil, "", fmt.Errorf("Error while extracting downloaded image: %v\n", err)
+	}
+
+	return inspectInfo, imageDigest, nil
+}
+
+// extractDownloadedImage will untar all the layer tar files specified in 'info'
+func (i *defaultImageInspector) extractDownloadedImage(layers []string) error {
+	for _, layer := range layers {
+		filename := i.opts.DstPath + "/" + strings.SplitN(layer, ":", 2)[1] + ".tar"
+		destDir := i.opts.DstPath
+		log.Printf("Untar %s\n", filename)
+		if err := util.UntarGzLayer(filename, destDir); err != nil {
+			return err
+		}
+	}
+	log.Printf("finished extraction")
+	return nil
+}
+
+// dockerCertPath will try to extract the registry name from the image and return
+// "/etc/docker/certs.d/<RERGISTRY_NAME>" if this path exists or nil otherwise.
+func (i *defaultImageInspector) certPath(fullImageName string) (string, error) {
+	if len(i.opts.RegistryCertPath) > 0 {
+		return i.opts.RegistryCertPath, nil
+	}
+
+	// try to find certificates from docker
+	source_name := strings.SplitN(fullImageName, "://", 2)
```

Comment:
i don't believe it is possible to have a panic here (even in the case of empty string, this function returns `[]string{""}` which won't cause a panic anywhere below